### DISCUSSION TITLE
Analyze samples to identify nuclei-free cells

### DISCUSSION THREAD

**Discussion by jashapiro** [AlexsLemonade/OpenScPCA-analysis#932] [Propose a new analysis]
### Proposed analysis

An intriguing paper by [Montserrat-Ayuso & Esteve-Codina (2024)](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-024-11015-5) suggests that many cells in current atlases are low quality by virtue of not containing nuclei. We should explore to what extent these concerns affect the samples in ScPCA.

### Scientific goals

We are always interested in improving the quality of the data in the atlas, and identifying nucleus-free cells is another layer of quality control that we may wish to add to the dataset.

### Methods or approach

As we do have both total and spliced counts for all genes, we should be able to calculate a proxy for the intronic content as the difference between those two. We could start by looking at the distribution of that statistic across cells; this would not be exactly the same as was done using the [`DropletQC` package](https://github.com/powellgenomicslab/DropletQC), as that requires access to the BAM files, but we should be able to do something comparable to identify cells with very low intronic fractions.

We should also look at expression of MALAT1 to see if the relationship observed by Montserrat-Ayuso & Esteve-Codina between that gene an intronic fraction is observed in our data.

### Existing modules

None.

### Input data

I would expect to start with the processed data, but it may be that we can obtain more clear results starting with the `filtered` data, where high mitochondrial content cells have not yet been removed. I would want to explore both to start. All analysis should be possible with the SCE objects.

### Scientific literature

https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-024-11015-5

### Other details

I expect this analysis to require minimal resources. Timeline TBD.

**Comment by sjspielman** [comment]
@jashapiro if you haven't started exploring here (which I gather you haven't since you're working on #945), I read through this paper earlier today and I'm definitely interested in picking this up! I see first steps as:
- Establish module (free step as a treat)
- Pick 1 library to explore a) intronic gene content, b) MALAT1 expression, and c) relationship between them. 
  - We should assess this for both the filtered and processed version of the library. For the filtered, we might also assess relationship to mitochondrial gene content. 
  - I'd be curious to see if clustering can pick up on any strong differences here, too.
  - For this PR, I expect some `utils` functions and an exploratory notebook in this PR
- We can then extend this initial exploration to a few more libraries (5ish?) where we have _a priori_ expectations of differing quality. With this comparison, we might be able to develop more context for which patterns we expect to see. 
  - We could run the exploratory notebook as a template these additional libraries 
  - I think it would be good to include both scRNA-seq and snRNA-seq examples here, too. We expect the "debris"-related quality issues raised in this paper to primarily affect scRNA-seq, so it would be useful to see if we can recover this pattern.
- From here, we should have a sense of what sort of/how much much there is there, and we can start thinking more specifically about how we might want to approach the full ScPCA, including what metrics would be worth calculating to provide to users.

**Reply by sjspielman** [reply]
I want to note the full top genes they identified as having high correlations with intronic content (or, nuclear fraction) from their table S1, specifically for the 10x data. Bonus, this might be a case for our new `rOpenScPCA` function to help convert ensembl -> gene symbol

- Tabula muris kidney 10x

  | Malat1 | Rbm39 | Pbx1 | Ddx5 | Macf1 | Foxp1
-- | -- | -- | -- | -- | -- | --
Nuclear fraction | 0.8290138 | 0.6845863 | 0.6234835 | 0.6230759 | 0.6216294 | 0.5973773

- MacParland liver

  | Malat1 | PRKCH | HLA-A | TMSB4X | PIP4K2A | SRGN
-- | -- | -- | -- | -- | -- | --
Nuclear fraction | 0.7768116 | 0.6369226 | 0.5989062 | 0.5969237 | 0.5937137 | 0.5884795

- Lukowski retina

  | MALAT1 | CACNB2 | CADM2 | CADM1 | EXOC4 | CAMK1D
-- | -- | -- | -- | -- | -- | --
Nuclear fraction | 0.9192181 | 0.8379755 | 0.7997234 | 0.7925404 | 0.791813 | 0.7812334

**Reply by jashapiro** [reply]
The base analysis seems straightforward enough that I would take a slightly different approach:

- Write a script to calculate stats (intronic content + normalized MALAT1 expression) for a library and output them as a table (which would probably include other `colData` like `sum` and `detected` and the `mito` stats).
- Read in those tables as an input to a report notebook that explores patterns across libraries.
- Explore some specific cases by reading in those tables + the original SCE objects.

I think a broad view of distributions of the stats across many samples will be important here, so I would prioritize doing a large number of libraries first. I expect the calculations to be quite fast as well.  It would still be straightforward to go back to look at specific examples for plotting and relationships to clustering results. 

I was somewhat equivocal in my earlier comment, but I think we definitely want to use `filtered` data as the input here, so you can also compare with the mitochondrial content including cells that would have been removed from the `processed` object.

**Reply by jashapiro** [reply]
> Bonus, this might be a case for our new `rOpenScPCA` function to help convert ensembl -> gene symbol

Not really, since you are going the opposite direction, and you know the genes you need.

I also don't see much value in going beyond MALAT1 for this analysis unless we are explicitly trying to define a better marker, which I would be a different phase of analysis.

**Reply by sjspielman** [reply]
> Not really, since you are going the opposite direction, and you know the genes you need.

This is....a good point 😞
